ecommerce/products/models.py

```diff
@@ -0,0 +1,14 @@
+from django.db import models
+
+# Create your models here.
+class Product(models.Model):
+    name = models.CharField(max_length=255)
+    product_type = models.CharField(max_length=255)
+    language = models.CharField(max_length=255)
+    rarity = models.CharField(default=None, blank=True, null=True, max_length=255)
+
+
+class ProductItem(models.Model):
+    product = models.ForeignKey('Product', on_delete=models.CASCADE)
+    condition = models.CharField(max_length=255)
```

Comment:
This looks really good so far. Couple things
(1) You don't need to set `default=None` on `rarity`. Not sure if it achieves anything but better off removing this.
(2) I think with `rarity`, `language`, `product_type`, `condition` you can use `choices`. This basically means your `CharField` only accepts values in a given set of choices. Have a look at how I've declared the `type` field on this model and see if you can work out how to use the same approach for your fields: https://github.com/johnpooch/diplomacy/blob/develop/core/models/piece.py

You'll also wanna look at https://github.com/johnpooch/diplomacy/blob/develop/core/models/base.py

Reply:
1) Had an error initially when doing this and stackoverflowed and the solution had that in it so just included it but wasn't really sure what it did
2) I'll look into that now